File: 3120-count-the-number-of-special-characters-i/3120-count-the-number-of-special-characters-i.cpp
class Solution {
public:
    int numberOfSpecialChars(string word) {
        unordered_set<char>upper;
        unordered_set<char>lower;
        int cnt=0;

        for(auto ch:word){
            if(isupper(ch)) upper.insert(tolower(ch));
            else lower.insert(ch);
        }

        for(auto ch:lower){
            if(upper.find(ch)!=upper.end()) cnt++;
        }
        return cnt;
    }
};

File: 3120-count-the-number-of-special-characters-i/3120-count-the-number-of-special-characters-i.java
class Solution {
    public int numberOfSpecialChars(String word) {
        Set<Character>upper=new HashSet<>();
        Set<Character>lower=new HashSet<>();
        int cnt=0;

        for(char ch:word.toCharArray()){
            if(Character.isUpperCase(ch)) upper.add(Character.toLowerCase(ch));
            else lower.add(ch);
        }

        for(char ch:lower){
            if(upper.contains(ch)) cnt++;
        }
        return cnt;
    }
}